File: Code/Core/Graphics/Platform/DX/SwapChain.cpp
#include "Graphics/Platform/DX/SwapChain.h"
#include <Windows.h>

Core::SwapChain::SwapChain(IDXGISwapChain* swapChain, uint32_t width, uint32_t height)
    : swapChain(nullptr)
    , width(width)
    , height(height)
{
}

void Core::SwapChain::Resize(uint32_t width, uint32_t height)
{
    this->width = width;
    this->height = height;
}

uint32_t Core::SwapChain::Width()
{
    return width;
}

uint32_t Core::SwapChain::Height()
{
    return height;
}

File: Code/WMFramework/Graphics/Private/Metal/SwapChain.m
//
//  SwapChain.m
//  WMFramework
//
//  Created by Eugene Kim on 2023/04/29.
//

#import <Foundation/Foundation.h>
#include "Graphics/Private/Metal/SwapChain.h"
 
using namespace WildMini;

SwapChain::SwapChain(CommandQueue* commandQueue, WMWindow* window)
{
}

SwapChain::~SwapChain()
{

}

const WMTexture* SwapChain::RenderTargetTexture() const
{
    return nullptr;
}

const WMTexture* SwapChain::DepthStencilTexture() const
{
    return nullptr;
}

void SwapChain::Resize(uint32_t width, uint32_t height)
{

}

void SwapChain::Present()
{ 
}
